python/api/job/serializers.py

```diff
@@ -52,23 +52,27 @@ def cook_obj(obj_type, obj_id, name):
         resp.append(cook_obj('cluster', obj.cluster.id, get_object_name(obj.cluster)))
         resp.append(cook_obj('service', obj.service.id, get_object_name(obj.service)))
     elif obj_type == 'host':
-        if task.action.host_action:
-            cluster = get_object_cluster(obj)
-            resp.append(cook_obj('cluster', cluster.id, get_object_name(cluster)))
-            if task.action.prototype.type == 'service':
-                service = ClusterObject.obj.get(cluster=cluster, prototype=task.action.prototype)
-                resp.append(cook_obj('service', service.id, get_object_name(service)))
-            elif task.action.prototype.type == 'component':
-                service = ClusterObject.obj.get(
-                    cluster=cluster, prototype=task.action.prototype.parent
-                )
-                component = ServiceComponent.obj.get(
-                    cluster=cluster, service=service, prototype=task.action.prototype
-                )
-                resp.append(cook_obj('service', service.id, get_object_name(service)))
-                resp.append(cook_obj('component', component.id, get_object_name(component)))
-        else:
-            resp.append(cook_obj('provider', obj.provider.id, get_object_name(obj.provider)))
+        if task.action:
```

Comment:
May be it would seems more straight-forward before line 49 smth like
if not task.action:
    return resp